validator/nodejs/index.js

```diff
@@ -142,6 +147,100 @@ function readFromUrl(url, userAgent) {
 }
 
 /**
+ * Creates a promise returning a list of crawled URLs beginning at the given
+ * start URL. Supported options are:
+ *
+ * <pre><code>
+ * {
+ *   crawlMaxDepth: 1,
+ *   crawlMaxConcurrency: 5,
+ *   userAgent: 'amphtml-validator'
+ * }
+ * </code></pre>
+ * @param {!string} startUrl
+ * @param {!Object} program
+ * @returns {!Promise<!string[]>}
+ */
+function crawlUrl(startUrl, program) {
+  return new Promise(function(resolve, reject) {
+    // Create and configure the crawler.
+    var crawler = new SimpleCrawler(startUrl);
+    crawler.maxDepth = program.crawlMaxDepth;
+    crawler.userAgent = program.userAgent;
+    crawler.maxConcurrency = program.crawlMaxConcurrency;
+
+    var discoveredAmpDocs = [];
+
+    // Queue AMPs for validation and discover outgoing links
+    crawler.discoverResources = function(buffer, queueItem) {
+      var body = buffer.toString('utf8');
+      var $ = cheerio.load(body);
+      // Discover any AMP doc
+      var ampDoc = null;
+      if (isAmp($.root())) {
+        // Doc itself is AMP
+        ampDoc = Promise.resolve(body);
+      } else {
+        // Doc references an AMP
+        var ampHref = $('link[rel=amphtml]').attr('href');
+        if (ampHref) {
+          ampDoc = readFromUrl(ampHref);
+        }
+      }
+      // Queue AMP doc for validation
+      if (ampDoc) {
+        discoveredAmpDocs.push(ampDoc);
+        // TODO remove hack to show URL in console
```

Comment:
Yes. That's what I meant by hack. Even worse: it doesn't work if you add multiple urls (sorry should've made this more explicit). 

I didn't include a fix in this PR as it means more changes of the whole codebase which would make this PR quite big.